site/components/pages/Pledge/PledgeDashboard/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from "react";
+import { ethers } from "ethers";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import { ButtonPrimary, Text } from "@klimadao/lib/components";
+import { concatAddress } from "@klimadao/lib/utils";
+
+import { Modal } from "components/Modal";
+import { Pledge } from "lib/moralis";
+
+import {
+  AssetBalanceCard,
+  AssetsOverTimeCard,
+  FootprintCard,
+  MethodologyCard,
+  PledgeCard,
+  RetirementsCard,
+} from "./Cards";
+import { PledgeForm } from "./PledgeForm";
+import { PledgeLayout } from "../PledgeLayout";
+import * as styles from "./styles";
+
+type Props = {
+  pageAddress: string;
+  pledge: Pledge;
+};
+
+const defaultValues = (pledge: Pledge): Pledge =>
```

Comment:
interesting pattern. If we just have strong types across the api endpoint and fetch code, do we need to do this?

Reply:
Potentially. Setting default values for the form/page removes the need to handle `{ pledge: null }`.
I'll defs revisit this when I get a clearer picture on the patterns we'd like to establish